src/qibo/transpiler/router.py

```diff
@@ -176,10 +176,13 @@ def __init__(
             self.circuit_blocks = blocks
         else:
             self.circuit_blocks = CircuitBlocks(circuit, index_names=True)
-        self.initial_layout = initial_layout
-        self._graph_qubits_names = [int(key[1:]) for key in initial_layout.keys()]
-        self._circuit_logical = list(range(len(initial_layout)))
-        self._physical_logical = list(initial_layout.values())
+        # Order the initial layout based on the hardware qubit name (this may create problems for a custom layout)
+        self.initial_layout = {
+            key: value for key, value in sorted(initial_layout.items())
+        }
+        self._graph_qubits_names = [int(key[1:]) for key in self.initial_layout.keys()]
+        self._circuit_logical = list(range(len(self.initial_layout)))
+        self._physical_logical = list(self.initial_layout.values())
```

Comment:
Do you mean that this is going to fail when the qubits names are not just `q0, q1, ...`? In that case I would at least raise a warning when `int_qubit_names == False` https://github.com/qiboteam/qibo/blob/c68e58bc2c0ba87a286f35544f97d69efab76d40/src/qibo/transpiler/pipeline.py#L193

Ideally, I think it would be better to internally always only support integer qubit identifiers, and, in case, to convert them to the user specified ones before returning them to any user facing layer.

Reply:
My comment on the code was not correct, it should say "ordering is necessary to avoid problems with custom layouts".
Now all the placers are creating an initial map using hardware qubit names "q0", "q1", "q2"... so there is no possibility to use an initial map with integer qubit names unless you define one by yourself without using a placer